src/zenml/utils/daemon.py

```diff
@@ -110,37 +112,197 @@ def daemon(*args: Any, **kwargs: Any) -> None:
     return inner_decorator
 
 
+def setup_daemon(
```

Comment:
This file is quite complicated to review because of all the moving functions. What I think the changes are:
- You extracted all the daemon setup into a separate function without major modification
- You moved most of the other functions outside to make them available on windows as well, without any major modification.

Is that correct @stefannica?

Reply:
Exactly!